File: cCarRentalSystem/Client/Domain/Models/Reservation.cs
namespace Domain.Models;

public class Reservation
{
    public int Id { get; set; }
    public string UserName{get;set;}
    public int CarId{get;set;}
    public DateTime StartDate{get;set;}
    public DateTime EndDate{get;set;}
    public double TotalPrice{get;set;}

    public bool IsCompleted { get; set; }
    public Reservation(int carId, string username, DateTime startDate, DateTime endDate, double totalPrice, bool isCompleted)
    {
        CarId = carId;
        UserName = username;
        StartDate = startDate;
        EndDate = endDate;
        TotalPrice = totalPrice;
        IsCompleted = isCompleted;
    }

    public Reservation(int id, string username, int carId, DateTime startDate, DateTime endDate, double totalPrice, bool isCompleted)
    {
        Id = id;
        UserName = username;
        CarId = carId;
        StartDate = startDate;
        EndDate = endDate;
        TotalPrice = totalPrice;
        IsCompleted = isCompleted;
    }

    public Reservation(){}

}

File: CarRentalService/src/main/java/com/example/carrentalservice/model/Reservation.java
package com.example.carrentalservice.model;

import com.google.type.DateTime;
import lombok.*;

import java.sql.Timestamp;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Reservation
{
    private int Id;
    private int CarId;
    private int UserId;
    private LocalDateTime StartDate;
    private LocalDateTime EndDate;
    private double TotalPrice;
}
